src/test/java/com/cloudbees/hudson/plugins/folder/FolderTest.java

```diff
@@ -278,6 +286,32 @@ private void copyFromGUI(Folder f, JenkinsRule.WebClient wc, String fromName, St
             }
         });
     }
+    
+    @Issue("JENKINS-32359")
+    @Test public void shouldProperlyPersistFolderPropertiesOnMultipleReloads() throws Exception {
+        Folder folder = r.jenkins.createProject(Folder.class, "myFolder");
+        r.jenkins.setAuthorizationStrategy(new ProjectMatrixAuthorizationStrategy());
+        
+        // We add a stub property to generate the persisted list
+        // After that we save and reload the config in order to drop PersistedListOwner according to the JENKINS-32359 scenario
+        folder.addProperty(new FolderCredentialsProvider.FolderCredentialsProperty(new DomainCredentials[0]));
+        folder.doReload();
```

Comment:
:bug: This is not convincing. You are retaining the local variable reference to the original `Folder` object, unlike the real situation where a new instance is constructed.

Rather use `Jenkins.reload` and call `getItemByFullName` to find the folder again; or use `RestartableJenkinsRule` in a new test suite.

Fixing this may also allow you to get rid of the distracting reference to `FolderCredentialsProperty`, which as @jtnord pointed out has no clear purpose in this test.